lib/bcchelper.py

```diff
@@ -0,0 +1,636 @@
+#
+# Copyright 2018, 2019 Delphix. All rights reserved.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+
+from bcc import BPF
+from time import sleep
+from time import time
+import ctypes as ct
+import logging
+import re
+
+""" BCC Helper
+This module contains the BCCHelper class to aid in writing BCC Tracing
+scripts using a python front end and bpf C code.  See the bcc Reference
+Guide.
+https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#1-bpf
+
+Defitions for C helper routines and macros are in
+/opt/delphix/server/etc/bcc_helper.h
+
+BCC Helper focuses are printing out data from a set of tracing aggregations.
+There are three supported scalar types of aggregations(count, sum, and
+aversum(which requires a separate count aggregation)), two histogram
+aggregations(log histogram or linear log histogram), and a stand alone
+average aggegation.  It is assumed that for each aggregation in the helper
+there has been implemented in C code a BPF_HASH using a common hash key.
```

Comment:
Should this comment make a stronger statement like `it is required that each aggregation in the bcc helper implement in C a BPF_HASH using a common key.`?

Reply:
Stronger is good.